### ISSUE TITLE
Issue With Transparent Sync Dependencies

### ISSUE THREAD

**Issue by williamsk** [kamsar/Unicorn#67]
In this project, I have two configs - one with transparent sync enabled and one without.  The one with transparent sync contains all of my templates.  The other one has some items that use those templates.  Whenever I try to sync the second config through the Unicorn control panel (this is an initial sync - the items do not yet exist in my database), it fails with an error that the template doesn't exist.  Although it does (it's in the transparent sync config and I can see a template with that ID in Sitecore).

Here's a snippet of the log from the control panel:

```
Once Configuration is being synced.
[A] master:/sitecore/content/XXX Website (6f40b18c-2f55-4ab5-8043-027987c995d5)
[A] master:/sitecore/content/XXX Website/home (37ea87c4-345c-478c-9aa7-730a6b91d0ec)
[A] master:/sitecore/content/XXX Website/shared-content (82246e6a-7c5d-4f30-b7f4-c7cad4ff0e12)
[A] master:/sitecore/content/XXX Website/site-settings (af91e57f-aa12-4efd-a42f-5084fe78ad59)

... this continues for a lot of items before the error occurs ...

281 errors occurred during deserialization. Earlier log messages for items below should be ignored; an error occurred that was retried and failed to correct.
master:/sitecore/content/XXX Website (6f40b18c-2f55-4ab5-8043-027987c995d5)
C:\Projects\XXX\src\items\Once\XXX Base Content\Content\XXX Website.yml
> DeserializationException: Template {9D585E79-FD65-4982-9BA6-DE8B6322E14A} for item /sitecore/content/XXX Website not found
master:/sitecore/content/XXX Website/home (37ea87c4-345c-478c-9aa7-730a6b91d0ec)
C:\Projects\XXX\src\items\Once\XXX Base Content\Content\XXX Website\home.yml
> ParentItemNotFoundException: Exception of type 'Sitecore.Data.Serialization.Exceptions.ParentItemNotFoundException' was thrown.
```

Template {9D585E79-FD65-4982-9BA6-DE8B6322E14A} does exist - it's part of the transparent sync config, and I can see it in Sitecore.

I have not been able to reproduce this with a super simple clean install, so it may be something I have configured incorrectly on this project or may be something special about one of the hundreds of items I'm trying to sync.  I will keep poking at a simple reproduction and add more info if I'm able to make it happen consistently outside of this one project.

**Comment by kamsar** [comment]
OMG.

![image](https://cloud.githubusercontent.com/assets/103677/10421255/c569026e-7059-11e5-9bbf-3e2c3516eef8.png)

**Comment by kamsar** [comment]
I thought this issue seemed really familiar as one that I'd fixed. Sure enough, I had. But I hadn't got the config change to activate the fix into the 3.0.0 config. #fail

See the commit diff, you won't need to upgrade to get this fixed as the code is in 3.0.0 just not activated.

**Comment by williamsk** [comment]
Thanks, @kamsar!  I just tested this with my original setup and that one config change definitely fixed the issue.
